File: OCP/ Kata_ Shops Types/SolucionJava/Shop/Persistence/ShopSQLPersistence.java
package SolucionJava.Shop.Persistence;

public class ShopSQLPersistence implements ShopSQLPersistenceInterface{
    //private SQLPersistence persistence;

    public ShopSQLPersistence(){
        //this.persistence = new SQLPersistence();
    }

    public void save(MiniShop shop){
        // Statement stmt = this.persistence.prepareQuery("INSERT INTO shops VALUES (?, ?)");
        // stmt.setString(1, shop.getName());
        // stmt.setInteger(2, shop.getPhone());
        // stmt.execute();
    }
}

File: ISP/Kata_ Shops Multirepository/SolucionPhp/Shop/Persistence/ShopSQLPersistence.php
<?php

declare(strict_types=1);

namespace SolucionPhp\Shop\Persistence;

use SolucionPhp\Shop\MiniShop;

class ShopSQLPersistence implements ShopSQLPersistenceInterface
{
    //private SQLPersistence $persistence;

    public function __construct()
    {
        //$this->persistence = new SQLPersistence();
    }

    public function save(MiniShop $shop): void
    {
        //$stmt = $this->persistence->prepare("INSERT INTO shops VALUES (:name, :phone)");
        /*$stmt->execute([
            'name' => $shop->getName(),
            'phone' => $shop->getPhone()
        ]);*/
    }
}
